integration_test/utils/test_util.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_config/flutter_config.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:package_info_plus/package_info_plus.dart';
+
+class TestUtil {
+  static Widget startAppWith(Widget widget) {
+    _initDependencies();
+    return MaterialApp(
```

Comment:
@manh-t I'm curious: what if we put our `MyApp()` to here instead of init from `MaterialApp`? 🤔  I think I have an idea why, but would like to hear from you

Reply:
@sleepylee actually we can put `MyApp()` to test. But it's not flexible enough when we need to test a specific screen or widget. This could lead to an issue that we have to modify the production code according to the test, which is not good imo.
Please correct me if I'm wrong, or guide me more if I miss any info 😓

Reply:
@manh-t but on the other hand, we can ensure that the test initial is the same as the real environment right? it makes the test fidelity & more reliable on the test result no? 🤔

Reply:
@sleepylee yes, it's more reliable for sure if the app is growing bigger and require more configs

Reply:
![image](https://user-images.githubusercontent.com/13435717/134621725-bb087861-696e-4e40-a1ee-20cf47eb5a95.png)

@manh-t ☝️ here is my concern: if the 'real' application has configured styling, when testing it doesn't display correctly on the test, so our test might not be very 'trustable', notice above: I changed the theme to `red` in the real App class and it doesn't reflect to the integration_test. Theme/locales... are just a part of a very long list of configuration:
![image](https://user-images.githubusercontent.com/13435717/134621874-71d589a0-0947-4e55-a65c-467ff5c45520.png)

and imagine that if our app has a lot of configurations, then our test might miss out the key points?

While your approach is still also valid in case we just want to test a small `Widget` without pumping the whole `RealApp()` - I think maybe we should clarify from the function we called? 

How about having 2 different test util method:

`pumpWidgetWithRealApp(widget: HomePage() or any page())`
and
`pumpWidgetWithShellApp(widget: HomePage() or any page())`

// Of course, put comments on each of them for the reader.

inside of the real MyApp() tho, we can modify a bit: allowing to have `home: Widget` inside the constructor; this will allow us to write test with any widget we want to pump directly, while not losing the clarity on the real code.

Moving further, I think if we implement deeplink later, allowing the App to pump a page directly will be somewhat helpful imo 🤔

Reply:
also notice that if you init with your `ShellApp()` - it doesn't have all the `navigation` path configured, if you want to to navigate I think it will be a trouble.

Reply:
actually deeplink in flutter is already done with the `path/to/navigate/` so yep - in case we want to test navigation with deeplink I think we will need to load all the path up

Reply:
> also notice that if you init with your `ShellApp()` - it doesn't have all the `navigation` path configured, if you want to navigate I think it will be a trouble.


@sleepylee yeah, because our app is still simple and does not have any other screen so I didn't add the navigation path. I'm thinking about setting up [fluro](https://pub.dev/packages/fluro) to support navigation (deeplink also). Maybe on `0.3.0` 💡 

> How about having 2 different test util method:
> 
> pumpWidgetWithRealApp(widget: HomePage() or any page())
> and
> pumpWidgetWithShellApp(widget: HomePage() or any page())


I agree with this, but I would like to modify a little bit. Because we often generate `routes` that include `/` as a root widget, we must not use `home: HomePage()` in `MaterialApp` as mentioned here.
![image](https://user-images.githubusercontent.com/60863885/134631164-99d66913-3a01-4d1e-b69b-f870e8f34f60.png)

So the `MyApp`'s constructor should be `MyApp(initialRoute: AppRoutes.HOME_PAGE)`. It's also valid when we implement the deeplink later. 😁

Reply:
the initialRoute would work for me yeah

Reply:
@sleepylee updated. Please review 🙋